File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.java
import java.util.*;

class Solution {
    public int solution(int[] bandage, int health, int[][] attacks) {
        
        int current_health = health;
        
        int count =1; //현재 시간

        for(int i=0; i < attacks.length; i++){
            
            int rest = attacks[i][0]-count-1;
            
//             System.out.println(rest);
            
            if(rest > 0){
                current_health += (int)rest/bandage[0] * bandage[2];
                current_health += (rest*bandage[1]);
            }
            
            if(current_health >= health){
                current_health = health;
            }
            
            count = attacks[i][0];
            current_health -= attacks[i][1];
                
            if(current_health <= 0){
                return -1;
            }
            
            // System.out.println(current_health + "b");
        }

        return current_health;
    }
}

File: 프로그래머스/1/250137. ［PCCP 기출문제］ 1번 ／ 붕대 감기/［PCCP 기출문제］ 1번 ／ 붕대 감기.js
function solution(bandage, health, attacks) {

    let current_health = health;
    let time = 1;
    
    for(let i of attacks){
        
        let count = i[0]-time-1;
        
        if(count > 0){
            current_health += parseInt((count/bandage[0])) * bandage[2];
            current_health += (count * bandage[1]);
        }
        
        if(current_health> health){
            current_health = health;
        }
        
        current_health -= i[1];
        
        if(current_health <= 0){
            return -1;
        }
        
        time = i[0];
    }
    
    return current_health;
}